src/components/auth_row/index.ts

```diff
@@ -0,0 +1,23 @@
+import template from './auth_row.hbs';
+import Block from '../../utils/Block';
+
+export interface AuthProps {
+    div_class?: string;
+    name: string;
+    type: string;
+    placeholder: string;
+}
+
+export class AuthRow extends Block<AuthProps> {
+    constructor(props: AuthProps) {
+        super('div', props);
+        if(this.element)  this.element.classList.add("auth__field")
+        if(props.div_class) {
+            if(this.element) this.element.classList.add(props.div_class)
+        }
```

Comment:
Смотри, тут три if'а, один из которых вложенный. Представь, что тебе надо будет дебажить этот код и каждый раз держать в голове, что у тебя есть if'ы. Это довольно сложно и этот кусок кода можно упростить

```suggestion
        if (!this.element) {
            return
        }

        this.element.classList.add("auth_field", props.div_class || '');
```